client/signup/steps/page-picker/use-cart-for-difm.tsx

```diff
@@ -0,0 +1,251 @@
+					lineCost: product.cost,
+					meta: translate( 'One-time fee' ),
+				};
+		}
+	} );
+}
+
+export function useCartForDIFM( selectedPages: string[] ): {
+	items: CartItem[];
+	total: string | null;
+	isCartLoading: boolean;
+} {
+	const signupDependencies = useSelector( getSignupDependencyStore );
+	const { newOrExistingSiteChoice, siteId } = signupDependencies;
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const proPlan = useSelector( ( state ) => getProductBySlug( state, PLAN_WPCOM_PRO ) );
+	const premiumPlan = useSelector( ( state ) => getProductBySlug( state, PLAN_PREMIUM ) );
+	const activePremiumPlanScheme = isEnabled( 'plans/pro-plan' ) ? proPlan : premiumPlan;
+
+	const extraPageProduct = useSelector( ( state ) =>
+		getProductBySlug( state, WPCOM_DIFM_EXTRA_PAGE )
+	);
+	const difmLiteProduct = useSelector( ( state ) => getProductBySlug( state, WPCOM_DIFM_LITE ) );
+
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const currentPlan = useSelector( ( state ) => getCurrentPlan( state, siteId ) );
+	const isPaidPlan = useSelector( ( state ) => isCurrentPlanPaid( state, siteId ) );
+
+	const cartKey = siteId;
+	const { replaceProductsInCart, responseCart, isLoading, isPendingUpdate } =
+		useShoppingCart( cartKey );
+
+	const difmExtra = useCallback(
+		() =>
+			buildDIFMCartExtrasObject( {
+				...signupDependencies,
+				selectedPageTitles: selectedPages,
+			} ),
+		[ signupDependencies, selectedPages ]
+	);
+
+	const shouldAddActivePremiumPlan =
+		currentPlan !== null && // Make sure current plan flag is loaded
+		! currentPlan && // Make sure there is no current plan
+		activePremiumPlanScheme &&
+		activePremiumPlanScheme.product_slug;
+
+	useEffect( () => {
+		siteId && dispatch( fetchSitePlans( siteId ) );
+		dispatch( requestProductsList() );
+	}, [ dispatch, siteId ] );
+
+	useEffect( () => {
+		if ( newOrExistingSiteChoice === 'existing-site' ) {
+			const productsToAdd: MinimalRequestCartProduct[] = [];
+			if ( shouldAddActivePremiumPlan ) {
+				productsToAdd.push( { ...activePremiumPlanScheme, extra: {} } );
+			}
+			if ( difmLiteProduct && difmLiteProduct.product_slug ) {
+				productsToAdd.push( {
+					...difmLiteProduct,
+					extra: difmExtra(),
+				} );
+			}
+
+			replaceProductsInCart( productsToAdd );
+		}
+	}, [
+		activePremiumPlanScheme,
+		difmLiteProduct,
+		newOrExistingSiteChoice,
+		replaceProductsInCart,
+		shouldAddActivePremiumPlan,
+		difmExtra,
+	] );
+
+	let displayedCartItems: CartItem[] = [];
+	if ( newOrExistingSiteChoice === 'existing-site' ) {
+		displayedCartItems = getSiteCartProducts( { responseCart, translate } );
+	} else {
+		displayedCartItems = getDummyCartProducts( {
+			selectedPages,
+			currencyCode: currencyCode ?? 'USD',
+			isPaidPlan,
+			formatCurrency,
+			translate,
+			activePlanScheme: activePremiumPlanScheme,
+			difmLiteProduct,
+			extraPageProduct,
+		} );
+	}
+
+	const totalCost = displayedCartItems.reduce(
```

Comment:
While testing the cart, I noticed two issues that will cause a discrepancy between the total calculated and the one shown on checkout. 
1. For certain geographies (e.g. US), taxes are also added to the total.
2. If a user has credits, the final total will be less.

Let's shift this discussion to pdh1Xd-Yw-p2 🙂

Reply:
There is a minimal cart version that already does not show these in the home page. So I think we should be fine.
![image](https://user-images.githubusercontent.com/3422709/176183326-6550a100-17cd-4cad-8781-38c95ef390c5.png)

Reply:
Yes, these calculations (and localizing the price) are often complicated and best left to the backend whenever possible.